lib/td/client/api/job.rb

```diff
@@ -287,6 +248,114 @@ def query(q, type=:hive, db=nil, result_url=nil, priority=nil, retry_limit=nil,
 
   private
 
+  def validate_content_length(response, current_total_chunk_size)
+    if expected_size = response.header['Content-Range'][0]
+      expected_size = expected_size[/\d+$/].to_i
+    elsif expected_size = response.header['Content-Length'][0]
+      expected_size = expected_size.to_i
+    end
+
+    if expected_size.nil?
+    elsif current_total_chunk_size < expected_size
+      # too small
+      raise EOFError, 'httpclient IncompleteError'
+    elsif current_total_chunk_size < expected_size
+      # too large
+      raise_error("Get job result failed", res)
+    end
+  end
+
+  def job_result_download(job_id, format='msgpack', decode_p=true)
+    client, header = new_client
+    client.send_timeout = @send_timeout
+    client.receive_timeout = @read_timeout
+    header['Accept-Encoding'] = 'deflate, gzip'
+
+    url = "/v3/job/result/#{e job_id}"
+    target = build_endpoint(url, @host)
+    params = {'format' => format}
+
+    unless ENV['TD_CLIENT_DEBUG'].nil?
+      puts "DEBUG: REST GET call:"
+      puts "DEBUG:   header: " + header.to_s
+      puts "DEBUG:   path:   " + target.to_s
+      puts "DEBUG:   params: " + params.to_s
+    end
+
+    # up to 7 retries with exponential (base 2) back-off starting at 'retry_delay'
+    retry_delay = @retry_delay
+    cumul_retry_delay = 0
+    current_total_chunk_size = 0
+    infl = nil
```

Comment:
This is ported from original code.
But [`chunk = infl.inflate(chunk) if infl`](https://github.com/treasure-data/td-client-ruby/pull/85/files#diff-799285ab8836ec49c3df0da1b6bb7735R322) can be killed by using `NullInflate`.
https://github.com/treasure-data/td-client-ruby/pull/85/files#diff-799285ab8836ec49c3df0da1b6bb7735R359
Is it better?